ash/src/extensions/khr/pipeline_executable_properties.rs

```diff
@@ -33,27 +32,15 @@ impl PipelineExecutableProperties {
         device: vk::Device,
         executable_info: &vk::PipelineExecutableInfoKHR,
     ) -> VkResult<Vec<vk::PipelineExecutableInternalRepresentationKHR>> {
-        let mut count = 0;
-        let err_code = self
-            .pipeline_executable_properties_fn
-            .get_pipeline_executable_internal_representations_khr(
-                device,
-                executable_info,
-                &mut count,
-                ptr::null_mut(),
-            );
-        if err_code != vk::Result::SUCCESS {
-            return Err(err_code);
-        }
-        let mut v: Vec<_> = vec![Default::default(); count as usize];
```

Comment:
Note that the `sType` in these structs is not initialized anymore which is probably the task of `get_pipeline_executable_internal_representations_khr` (etc. below). This is likely fine to assume but I want to prevent a situation like #394.

Reply:
> which is probably the task of get_pipeline_executable_internal_representations_khr (etc. below).

Unfortunately, this isn't true. We *must* initialize both `sType` and, even more importantly, `pNext` correctly in every case. The implementation may traverse `pNext` in search of extension output structures. I recommend introducing a variant of `read_into_vector` which takes an initial value, or perhaps which invokes `T::default()` internally.

Reply:
That's horrible, but true for _every_ function returning its result in a "base" structure (with `sType`/`pNext` fields). Is there anything we can automate with traits here (or at least safeguard), maybe something with a `BaseOutStructure`-like trait the generator could emit?

Reply:
We could have an automated (private) trait telling how to initialize the given datastructure were it used in an output position: uninitialized for handles and the like, `Default::default()` for `BaseOutStructure`-like types.

Alternatively we can take the performance "hit" of always `default`-initializing every vector instead of using `with_capacity()` and `unsafe fn set_len()`.

Finally we can have two distinct functions for this, but it's up to authors and reviewers to ensure the right variant is used. If so, I'd make the more-unsafe/unreliable uninitialized variant obvious in the name (`read_into_uninitialized_vector`) instead of the saner/safer variant (`read_into_defaulted_vector`).

Reply:
In openxrs, I generate associated functions of this form:
```rust
    pub fn out(next: *mut BaseOutStructure) -> MaybeUninit<Self> {
        let mut x = MaybeUninit::<Self>::uninit();
        unsafe {
            (x.as_mut_ptr() as *mut BaseOutStructure).write(BaseOutStructure {
                ty: Self::TYPE,
                next,
            });
        }
        x
    }
```
To avoid adding large amounts of code, this could be a single `unsafe` function that assumes a suitable `T`. Alternatively, we could add an empty `unsafe trait OutStructure` and make the function itself safe. I'm not sure if this is really worth the additional complexity over just using `Default` (even C++ Vulkan developers tend to 0-initialize structures wholesale as a relatively sane default) but it should work fine. However, perhaps it would be best to leave that to a followup PR?

Reply:
@Ralith I suppose we could generate this function for exactly every structure that receives `.push_next` (unless there are structs used in this out-position with no known extension yet, while obviously still requiring appropriate initialization), and I have some ideas to move all that duplication to a single definition with traits. We can use the same trait bound for `fn push_next()` in `fn out()` as well.

We will have the `MaybeUninit<Self>`'s in vecs/slices to pass them at-large to Vulkan though, how are you converting them back to their initialized form in openxrs? `transmute`?

Either way, let's leave this for a followup PR. The initial default-initialization for `pipeline_executable_properties.rs` has been restored. I went over all the calls to `read_into_uninitialized_vector` and found only `get_physical_device_tool_properties` which was (even before this PR) not performing the appropriate initialization.

PS. I like the idea of keeping everything but `s_type`/`p_next` uninitialized!